h2o-py/h2o/explanation/_explain.py

```diff
@@ -1302,7 +1307,41 @@ def _has_varimp(model):
     :param model: model or a string containing model_id
     :returns: bool
     """
-    return _get_algorithm(model) not in ["stackedensemble", "naivebayes"]
+    if isinstance(model, h2o.model.ModelBase):
+        # check for cases when variable importance is disabled or
+        # when a model is stopped sooner than calculating varimp (xgboost can rarely have no varimp).
+        output = model._model_json["output"]
+        return output.get("variable_importances") is not None
+    else:
+        return _get_algorithm(model) not in ["stackedensemble", "naivebayes"]
```

Comment:
Cool, I didn't know that!

The problem is that sometimes some models don't have varimp even though the algo supports it, e.g., XGBoost that is stopped sooner than calculating varimp - that's why there is the `output.get("variable_importances") is not None` to check that there really are some results.

The other branch is using `model_id` - if the prefix of the model id is a known algo (`ALGONAME_.*`) use that (transformed to the same name as is the `model.algo`), otherwise `h2o.get_model(model_id).algo`.

The `has_extension(model, 'VariableImportance')` would require me to `get_model`s in order to find if they have varimp and I think it could give me false positives - seems that the mixin is added without looking if the model really has the varimp data.

Reply:
> The other branch is using model_id - if the prefix of the model id is a known algo (ALGONAME_.*) use that (transformed to the same name as is the model.algo), otherwise h2o.get_model(model_id).algo

We should always avoid using `model_id` to extract information about the model: `model_id` is often entered by the user, and we can't trust the user (copy/paste, change the estimator…).
So yes, you should only rely on the result from `get_model`, and you need to ensure that `get_model` is always called once, i.e never keep ids further that the main entry point.

Reply:
> that's why there is the output.get("variable_importances") is not None to check that there really are some results

you can keep this additional condition. Was mainly concerned with the algo check:
```return _get_algorithm(model) not in ["stackedensemble", "naivebayes"]```